### PR TITLE
Bug 1907792: Change the OperatorUpgradeable condition to Upgradeable

### PR DESCRIPTION
**Pull request by dinhxuanvu**
The original design intends to use the name `Upgradeable` only.

Signed-off-by: Vu Dinh <vdinh@redhat.com>

**Description of the change:**

Gate operator upgrade on the "Upgradeable" OperatorCondition status type instead of "OperatorUpgradeable".

**Motivation for the change:**

The [OperatorConditions enhancement](https://github.com/operator-framework/enhancements/blob/master/enhancements/operator-conditions.md#example-upgradeable-condition) indicates that OLM gates upgrades on a status condition type of "Upgradeable", not "OperatorUpgradeable".

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

**Comment by dinhxuanvu** [comment]
/hold
until this #1903 gets merged first

File: test/e2e/operator_condition_e2e_test.go
Comment (awgreene) [inline test/e2e/operator_condition_e2e_test.go:28]:
```suggestion
			"Upgrade condition type is set in OperatorCondition CR. Plus, an operator" +
```

**Review by njhale — APPROVED** [review]
I think we got lucky that we didn't cut a release -- in both upstream and OCP -- when this was an option. If we had and we were being strict about it, I think we would have needed to continue support for "OperatorUpgradeable" until we could formally deprecate it.

/approve

**Pull request merged**
